File: c/grains/grains.c
#include "grains.h"


uint64_t square(uint8_t index)
{
    int i = index-1;
    if(index > 1 || index < 64){
        return pow(2, i);
    }
    return 0;
}
uint64_t total(void)
{
    int r = 0;
    for (int i = 0; i < 64; i++)
    {
        r += square(i);
    }
    return r;
}

File: csharp/grains/Grains.cs
public static class Grains
{
    public static ulong Square(int n)
    {
        if (n > 64 || n < 1)
        {
            throw new ArgumentOutOfRangeException();
        }
        return (ulong)Math.Pow(2, n - 1);
    }

    public static ulong Total()
    {
        ulong tot = 0;
        int n = 64;
        while (n > 0)
        {
            tot += Square(n);
            n--;
        }
        return tot;
    }
}
